lib/transaction/trace/index.js

```diff
@@ -84,14 +84,24 @@ Trace.prototype.generateSpanEvents = function generateSpanEvents() {
     const spanAggregator = this.transaction.agent.spanEventAggregator
 
     const children = this.root.getChildren()
-    
-    // At the point where 'host.displayName' is available we only 
-    // have a root span.  Adding attribute to first non-root span here.
-    if (children.length > 0 && this.displayName) {
-      children[0].addSpanAttribute(
-        'host.displayName',
-        this.displayName
-      )
+
+    if (children.length > 0) {
+      // At the point where these attributes are available, we only have a
+      // root span. Adding attributes to first non-root span here.
+      const attributeMap = {
+        'host.displayName': this.displayName,
+        'parent.type': this.transaction.parentType,
+        'parent.app': this.transaction.parentApp,
+        'parent.account': this.transaction.parentAcct,
+        'parent.transportType': this.transaction.parentTransportType,
+        'parent.transportDuration': this.transaction.parentTransportDuration
+      }
+
+      for (const [key, value] of Object.entries(attributeMap)) {
```

Comment:
I suppose it cleans up the code a bit here but feels a bit unnecessary to generate the entries for iterating when we already have to provide a key and provide a value via accessor for the key/value pairs. 

should only happen once per transaction, so impact is low. not going to impact PR approval.